user-operator/src/main/java/io/strimzi/operator/user/UserOperatorConfig.java

```diff
@@ -14,11 +13,106 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Properties;
+import java.util.HashMap;
+import java.util.Objects;
+import java.util.Set;
+import java.util.Collections;
 
 /**
  * Cluster Operator configuration
  */
 public class UserOperatorConfig {
+
+
+    private static abstract class Type<T> {
+        abstract T parse(String s);
```

Comment:
As an abstraction this probably needs to define:

1. what exception is thrown if `s` cannot be parsed
2. whether the result can ever be null

Reply:
Yet to be fixed